packages/components/src/card/types.ts

```diff
@@ -0,0 +1,81 @@
+/**
+ * External dependencies
+ */
+// eslint-disable-next-line no-restricted-imports
+import type { CSSProperties } from 'react';
+
+/**
+ * Internal dependencies
+ */
+import type { Props as SurfaceProps } from '../surface/types';
+
+export type SizeOptions = 'xSmall' | 'small' | 'medium' | 'large';
+
+type SizeableProps = {
+	/**
+	 * Determines the amount of padding within the component.
+	 *
+	 * @default 'medium'
+	 */
+	size?: SizeOptions;
+};
+
+export type Props = SurfaceProps &
+	SizeableProps & {
+		/**
+		 * Renders without a border.
+		 *
+		 * @default false
+		 */
+		isBorderless?: boolean;
+		/**
+		 * Renders with rounded corners.
+		 *
+		 * @default true
+		 */
+		isRounded?: boolean;
+		/**
+		 * Renders with elevation styles (box shadow).
+		 *
+		 * @default false
+		 */
+		isElevated?: boolean;
+	};
+
+type BaseSubComponentProps = SizeableProps & {
+	/**
+	 * The children elements.
+	 */
+	children: React.ReactNode;
+	/**
+	 * Renders with a light gray background color.
+	 *
+	 * @default false
+	 */
+	isShady?: boolean;
+};
+
+export type BodyProps = BaseSubComponentProps & {
+	/**
+	 * Determines if the component is scrollable.
+	 *
+	 * @default true
+	 */
+	isScrollable?: boolean;
+};
+
+type MarginalSubComponentProps = BaseSubComponentProps & {
```

Comment:
Nit: I'd likely name that `CardSubComponentProps`, but I don't have strong feelings on the naming.

Reply:
> Nit: I'd likely name that CardSubComponentProps

Do you refer to `MarginalSubComponentProps` or `BaseSubComponentProps`?

Reply:
I'm going to go ahead and merge this PR. If we agree on any changes, I will apply them in the follow-up PR #32566

Reply:
Note taken, let's keep naming as it is then!